scripts/migration/migrate_twitter_handles.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import sys
+
+from modularodm import Q
+
+from framework.transactions.context import TokuTransaction
+from nose.tools import *
+from tests.base import OsfTestCase
+from tests.factories import UserFactory
+from website.app import init_app
+from website.models import User
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+def get_targets():
+    return User.find(Q('social.twitter', 'contains', '@'))
+
+def migrate(targets, dry_run=True):
+    # iterate over targets
+    # log things
+    users = targets
+    logging.info(len(users))
+    for user in users:
+        twitter = user.social['twitter'].replace("@", "")
+        logger.info('Setting `social.twitter` field of user {0} to {1}'.format(user._id, twitter))
+        if not dry_run:
+            user.social['twitter'] = twitter
+            user.save()
+    
+    if dry_run:
+        raise RuntimeError('Dry run, transaction rolled back.')
+
+
+def main():
+    dry_run = False
+    if '--dry' in sys.argv:
+        dry_run = True
+    if not dry_run:
+        script_utils.add_file_logger(logger, __file__)
+    init_app(set_backends=True, routes=False)
+    with TokuTransaction():
+        migrate(targets=get_targets(), dry_run=dry_run)
+
+if __name__ == "__main__":
+    main()
+
+class TestMigrateTwitterHandles(OsfTestCase):
+    def setUp(self):
+        super(TestMigrateTwitterHandles, self).setUp()
+        # User 2 has no '@' signs, should be unaffected by migration
+        self.user1 = UserFactory()
+        self.user1.social['twitter'] = 'user1'
+        self.user1.save()
+        # User 2 has 1 leading '@', should be changed to 'user2' after migration
+        self.user2 = UserFactory()
+        self.user2.social['twitter'] = '@user2'
+        self.user2.save()
+        # User 3 has 2 leading '@', should be changed to 'user1' after migration
+        self.user3 = UserFactory()
+        self.user3.social['twitter'] = '@@user3'
+        self.user3.save()
+        # User 4 has many leading '@' signs, should be changed to 'user4' after migration
+        self.user4 = UserFactory()
+        self.user4.social['twitter'] = '@@@@@@@@@@@@@@@@@user4'
+        self.user4.save()
+        # User 5 has '@' signs in the middle of their handle, should be changed to 'user5' after migration
+        self.user5 = UserFactory()
+        self.user5.social['twitter'] = 'user@@@@@@@5'
```

Comment:
Is there a reason to remove non-leading "@"s? Are you certain that these aren't valid. Seems safer to only strip leading "@"'s anyway.